Source/SwiftLintFramework/Rules/Style/SortedEnumCasesRule.swift

```diff
@@ -0,0 +1,123 @@
+import SwiftSyntax
+
+struct SortedEnumCasesRule: ConfigurationProviderRule, SwiftSyntaxRule, OptInRule {
+    var configuration = SeverityConfiguration(.warning)
+
+    init() {}
+
+    static let description = RuleDescription(
+        identifier: "sorted_enum_cases",
+        name: "Sorted Enum Cases",
+        description: "Enum cases should be sorted",
+        kind: .style,
+        nonTriggeringExamples: [
+            Example("""
+            enum foo {
+                case a
+                case b
+                case c
+            }
+            """),
+            Example("""
+            enum foo {
+                case a, b, c
+            }
+            """),
+            Example("""
+            enum foo {
+                case a
+                case b, c
+            }
+            """),
+            Example("""
+            enum foo {
+                case a(foo: Foo)
+                case b(String), c
+            }
+            """),
+            Example("""
+            @frozen
+            enum foo {
+                case b
+                case a
+                case c
+                case c, f, d
+            }
+            """).focused()
```

Comment:
```suggestion
            """)
```
This fails the builds.